### PR TITLE
Code base cleanup

### PR DESCRIPTION
**Pull request by pith**
Clean up the kernel in order to prepare the coming new feature (from the PEPs).

- Various code simplification (without behavior changes)
- Remove commented code
- Fix typos
- Extract inner classes from the KernelCore
- Change some log level from info to debug
- Add logback-test.xml in tests to remove Reflections' stack trace in the logs

@ejemba: I'm sorry, I now there are too many changes, so the PR might be hard to review. But I hope you will like it ;). Tell me if you need some changes/rollback.

### CODE REVIEW

**Comment by pith** [comment]
@ejemba I rebased this PR. It can now be merged if it's OK for you.

File: core/src/main/java/io/nuun/kernel/core/internal/KernelCore.java [hunk 1]
```diff
@@ -303,20 +279,19 @@ private void checkPlugins()
...
-                    // if (kernelParams.keySet().containsAll(computedMandatoryParams))
                     {
                         pluginClasses.add(plugin.getClass());
                     }
                     else
                     {
-                        logger.error("plugin {} miss parameter/s : {}", pluginName, kernelParamsRequests.toString());
-                        throw new KernelException("plugin " + pluginName + " miss parameter/s : " + kernelParamsRequests.toString());
+                        logger.error("Plugin {} misses parameter/s : {}", pluginName, kernelParamsRequests.toString());
+                        throw new KernelException("Plugin " + pluginName + " miss parameter/s : " + kernelParamsRequests.toString());
```
Comment (ejemba) [inline core/src/main/java/io/nuun/kernel/core/internal/KernelCore.java]: « miss → misses » as well, If I am correct. Like you did the line above.

**Comment by ejemba** [comment]
This looks good to me, only one « miss → misses »
and it should be ok.

**Comment by pith** [comment]
Good catch. I fixed it and made some other javadoc improvements

**Pull request merged**